pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -1123,8 +1123,25 @@ func (b *MasqueradeBindMechanism) createNatRulesUsingNftables(proto iptables.Pro
 	if len(b.iface.Ports) == 0 {
 		err = Handler.NftablesAppendRule(proto, "nat", "KUBEVIRT_PREINBOUND",
 			"counter", "dnat", "to", b.getVifIpByProtocol(proto))
+		if err != nil {
+			return err
+		}
 
-		return err
+		err = Handler.NftablesAppendRule(proto, "nat", "KUBEVIRT_POSTINBOUND",
+			Handler.GetNFTIPString(proto), "daddr", b.getVifIpByProtocol(proto),
+			"counter", "snat", "to", b.getGatewayByProtocol(proto))
```

Comment:
Hello @alonSadan , sorry for being back and forth on this. We realized one bad consequence of using the destination IP address for matching in `KUBEVIRT_POSTINBOUND`.
This rule matches traffic going into the VM from k8s pod network, so the VM sees the origin as 10.0.2.1.
Since we want he masquarading to be transparent for the VM, we don't want to change the source IP for traffic coming from the pod network.
Could you please change the rules back to what you had initially?

Reply:
Done.